Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs

```diff
@@ -943,27 +928,392 @@ public unsafe TValue ReadValue<TValue>()
+        ///     StartFiring();
+        /// else if (fire.WasReleasedThisFrame())
+        ///     StopFiring();
+        /// </code>
+        /// </example>
+        ///
+        /// This method will disregard whether the action is currently enabled or disabled. It will keep returning
+        /// true for the duration of the frame even if the action was subsequently disabled in the frame.
+        /// </remarks>
+        /// <seealso cref="IsPressed"/>
+        /// <seealso cref="WasPressedThisFrame"/>
+        /// <seealso cref="CallbackContext.ReadValueAsButton"/>
+        /// <seealso cref="WasPerformedThisFrame"/>
+        public unsafe bool WasReleasedThisFrame()
+        {
+            var state = GetOrCreateActionMap().m_State;
+            if (state != null)
+            {
+                var actionStatePtr = &state.actionStates[m_ActionIndexInState];
+                var currentUpdateStep = InputUpdate.s_UpdateStepCount;
+                return actionStatePtr->releasedInUpdate == currentUpdateStep && currentUpdateStep != default;
+            }
+
+            return false;
+        }
+
+        ////REVIEW: Should we also have WasStartedThisFrame()? (and WasCanceledThisFrame()?)
+
+        /// <summary>
+        /// Check whether <see cref="phase"/> was <see cref="InputActionPhase.Performed"/> at any point
+        /// in the current frame.
+        /// </summary>
+        /// <returns>True if the action performed this frame.</returns>
+        /// <remarks>
+        /// This method is different from <see cref="WasPressedThisFrame"/> in that it depends directly on the
+        /// interaction(s) driving the action (including the default interaction if no specific interaction
+        /// has been added to the action or binding).
+        ///
+        /// For example, let's say the action is bound to the space bar and that the binding has a
+        /// <see cref="Interactions.HoldInteraction"/> assigned to it. In the frame where the space bar
+        /// is pressed, <see cref="WasPressedThisFrame"/> will be true (because the button/key is now pressed)
+        /// but <c>WasPerformedThisFrame</c> will still be false (because the hold has not been performed yet).
+        /// Only after the hold time has expired will <c>WasPerformedThisFrame</c> be true and only in the frame
+        /// where the hold performed.
+        ///
+        /// This is different from checking <see cref="phase"/> directly as the action may have already progressed
+        /// to a different phase after performing. In other words, even if an action performed in a frame, <see cref="phase"/>
+        /// may no longer be <see cref="InputActionPhase.Performed"/> whereas <c>WasPerformedThisFrame</c> will remain
+        /// true for the entirety of the frame regardless of what else the action does.
+        ///
+        /// In the same vein, unlike <see cref="ReadValue{TValue}"/>, which will reset when the action goes back to waiting
```

Comment:
```suggestion
        /// Unlike <see cref="ReadValue{TValue}"/>, which will reset when the action goes back to waiting
```

Reply:
Style: Microsoft and others advise to avoid using metaphors, similes and other expressions that might be perceived as ambiguous by ESL and other types of users with reading difficulties.

Reply:
Pushed